addons/mail/static/src/messaging/component/activity/activity_tests.js

```diff
@@ -770,14 +770,117 @@ QUnit.test('activity click on edit', async function (assert) {
     assert.strictEqual(
         document.querySelectorAll('.o_Activity_editButton').length,
         1,
-        "should have activity mail template name preview button"
+        "should have activity edit button"
     );
 
     document.querySelector('.o_Activity_editButton').click();
     await afterNextRender();
     assert.verifySteps(
         ['do_action'],
-        "should have call 'schedule activity' action correctly"
+        "should have called 'schedule activity' action correctly"
+    );
+});
+
+QUnit.test('activity edition', async function (assert) {
+    assert.expect(15);
+
+    await this.start({
+        intercepts: {
+            do_action(ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_id,
+                    42,
+                    'Action should have the activity res id as default res id in context'
+                );
+                assert.strictEqual(
+                    ev.data.action.context.default_res_model,
+                    'res.partner',
+                    'Action should have the activity res model as default res model in context'
+                );
+                assert.strictEqual(
+                    ev.data.action.type,
+                    'ir.actions.act_window',
+                    'Action should be of type "ir.actions.act_window"'
+                );
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    'mail.activity',
+                    'Action should have "mail.activity" as res_model'
+                );
+                assert.strictEqual(
+                    ev.data.action.res_id,
+                    12,
+                    'Action should have activity id as res_id'
+                );
+                ev.data.options.on_close();
+            },
+        },
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/activity_format') {
+                assert.step('activity_format');
+                return [{
+                    can_write: true,
+                    icon: 'fa-check',
+                    id: 12,
+                    res_id: 42,
+                    res_model: 'res.partner',
+                }];
```

Comment:
maybe define `mockedActivityFormat` in `mail/mock_server`, and make use of data instead?

and modify icon from intercepted `do_action`